src/cmakeWorkspaceFolder.ts

```diff
@@ -207,37 +47,85 @@ export class CMakeWorkspaceFolderController implements vscode.Disposable {
     }
 
     /**
-     * The active workspace folder. This controls several aspects of the extension,
-     * including:
-     *
-     * - Which CMakeProject backend receives commands from the user
-     * - Where we search for variants
-     * - Where we search for workspace-local kits
+     * The path of the folder of the active CMakeProject instance
      */
-    private _activeFolder?: CMakeWorkspaceFolder;
-    get activeFolder() {
-        return this._activeFolder;
+    get activeFolderPath(): string | undefined {
+        return this.activeProject?.folderPath;
     }
-    setActiveFolder(ws: vscode.WorkspaceFolder | undefined) {
-        if (ws) {
-            this._activeFolder = this.get(ws);
-        } else {
-            this._activeFolder = undefined;
+    /**
+     * The name of the folder of the active CMakeProject instance
+     */
+    get activeFolderName(): string | undefined {
+        return this.activeProject?.folderName;
+    }
+
+    private activeProject: CMakeProject | undefined;
+    updateActiveProject(workspaceFolder?: vscode.WorkspaceFolder, openEditor?: vscode.TextEditor, folderName?: string): void {
+        if (!workspaceFolder && !openEditor && !folderName) {
+            return;
+        }
+        // When the folderName is defined, it means the active CMake project is being updated from the status bar.
+        // In this case, we need to search for the new project in the list of all projects.
+        if (!workspaceFolder && !openEditor && folderName) {
```

Comment:
I think adding `folderName` as a parameter to this function makes things less clear to callers of this function.  We should have a verison of this function that takes `workspaceFolder + openEditor` and another version that just takes `folderName` because you ignore the other parameters if one of either set is used.

The `folderName` one can be called `updateActiveProjectByName` or something.